orderbook/src/account_balances.rs

```diff
@@ -143,15 +151,81 @@ impl BalanceFetching for Web3BalanceFetcher {
         };
         let _ = self._register_many(subscriptions.into_iter()).await;
     }
+
+    async fn can_transfer(&self, token: H160, from: H160, amount: U256) -> bool {
+        let instance = ERC20::at(&self.web3, token);
+        let calldata = instance
+            .transfer_from(from, self.settlement_contract, amount)
+            .tx
+            .data
+            .unwrap();
+        let call_request = CallRequest {
+            from: Some(self.allowance_manager),
+            to: Some(token),
+            data: Some(calldata),
+            ..Default::default()
+        };
+        let block = Some(BlockId::Number(BlockNumber::Latest));
+        let response = self.web3.eth().call(call_request, block).await;
+        response
+            .map(|bytes| is_empty_or_nonzero(bytes.0.as_slice()))
+            .unwrap_or(false)
+    }
+}
+
+fn is_empty_or_nonzero(bytes: &[u8]) -> bool {
```

Comment:
ubernit: *feel free to disregard* - I would call this
```suggestion
fn is_empty_or_truthy(bytes: &[u8]) -> bool {
```

Since it does not accept any non-zero byte slice.